src/Groups/matrices/stuff_field_gen.jl

```diff
@@ -0,0 +1,52 @@
+import Hecke: evaluate, field_extension, FinField, FinFieldElem, PolyElem
+
+export primitive_element
+
+function _change_type(f::PolyElem{T}) where T <: FinFieldElem
+   e,p = ispower(order(base_ring(f)))
+   F = GF(Int(p),Int(e))[1]
+   t = PolynomialRing(F,"t")[2]
+   return sum([t^i*F(lift(coeff(f,i))) for i in 0:degree(f)])
+end
+
+# return a generator for the unit group of F
+function _centralizer(f::PolyElem{T}) where T <: FinFieldElem
+  if typeof(f)!=fq_nmod_poly && typeof(f)!=fq_poly
+     f = _change_type(f)
+  end
+  K = base_ring(f)
+  d = degree(K)
+  e = degree(f)
+  q = order(K)
+  L, mL = field_extension(f)
+#  m = divexact(order(L)-1, order(K)-1)
+#  U, mU = unit_group(L, n_quo = Int(m))
```

Comment:
```suggestion
```

Reply:
As I commented above, these two lines instead of the line below `U, mU = unit_group(L)` will produce a generator for ` F* / (prime subfield)*` at a much lower cost. Anyway, Carlo told me that he will soon have a faster procedure, so this function may change in future.
Do you think I can remove the comments since they create just confusion?

Reply:
I think such commented out code should either be removed; or if there is a reason to keep it around, then that reason should be in a comment right before the commented out code.

Also, the information that this method is inefficient and "Carlo will soon add a faster procedure somewhere" (where?) is also something that should be in a comment on the function, so that this knowledge is not lost / forgotten.

Reply:
Here's the thing: you write before the function:
> removing the three commented lines, the output is a generator for F / its prime subfield

But so what?! It is not explained what any of that would matter, so that comment as well as then commented out code are useless. Once you are gone, nobody will have a clue why any of this matters. 

So either explain why it matters, or else remove it?

Reply:
Ok, I just remove them.